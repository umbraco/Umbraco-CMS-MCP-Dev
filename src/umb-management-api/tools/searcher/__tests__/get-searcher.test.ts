import GetSearcherTool from "../get/get-searcher.js";
import { UmbracoManagementClient } from "@umb-management-client";
import { jest } from "@jest/globals";

describe("get-searcher", () => {
  let originalConsoleError: typeof console.error;
  let originalGetClient: typeof UmbracoManagementClient.getClient;

  beforeEach(() => {
    originalConsoleError = console.error;
    console.error = jest.fn();
    originalGetClient = UmbracoManagementClient.getClient;
  });

  afterEach(() => {
    console.error = originalConsoleError;
    UmbracoManagementClient.getClient = originalGetClient;
  });

  it("should list all searchers with default parameters", async () => {
    const result = await GetSearcherTool().handler(
      { take: 100 },
      { signal: new AbortController().signal }
    );
    // Verify the handler response using snapshot
    expect(result).toMatchSnapshot();
  });

});