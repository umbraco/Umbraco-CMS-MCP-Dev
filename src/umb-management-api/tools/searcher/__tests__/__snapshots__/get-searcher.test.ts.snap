// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get-searcher should list all searchers with default parameters 1`] = `
{
  "content": [
    {
      "text": "{"total":1,"items":[{"name":"MultiSearcher"}]}",
      "type": "text",
    },
  ],
}
`;
